TodoList.Application/CurrentUserProvider.cs

```diff
@@ -1,11 +1,43 @@
-﻿namespace TodoList.Application
+﻿using Microsoft.AspNetCore.Http;
+using System.Security.Claims;
+using TodoList.Application;
+
+public class CurrentUserProvider : ICurrentUser
 {
-    public class CurrentUserProvider : ICurrentUser
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public CurrentUserProvider(IHttpContextAccessor httpContextAccessor)
+    {
+        _httpContextAccessor = httpContextAccessor;
+    }
+
+    public string Id
     {
-        public int Id => 99;
+        get
+        {
+            var userId = _httpContextAccessor.HttpContext.User.FindFirst(ClaimTypes.NameIdentifier)?.Value;
+            if (userId != null)
+            {
+                return userId;
+            }
 
-        public string Name => "Test User";
+            return string.Empty;
```

Comment:
If some fields are not present or user is not logged in, exception should be thrown. Never ignore such cases, what will be if user don't have an ID?